pkg/builder/builder.go

```diff
@@ -120,7 +120,50 @@ func (b generalBuilder) NumGPUs() int {
 	return ir.NumGPUs()
 }
 
+// Always return updated when met error
+func (b generalBuilder) CheckDepsFileUpdate(ctx context.Context, tag string, deps []string) (bool, error) {
+	dockerClient, err := docker.NewClient(ctx)
+	if err != nil {
+		return true, err
+	}
+	image, err := dockerClient.GetImage(ctx, tag)
+	if err != nil {
+		return true, err
+	}
+	imageCreatedTime := image.Created
+
+	latestTimestamp := int64(0)
+	for _, dep := range deps {
+		file, err := os.Stat(dep)
+		if err != nil {
+			return true, err
+		}
+		modifiedtime := file.ModTime().Unix()
+		// Only needt o use the latest modified time
+		if modifiedtime > latestTimestamp {
+			latestTimestamp = modifiedtime
+		}
+	}
+	if latestTimestamp > imageCreatedTime {
+		return true, nil
+	}
+	return false, nil
+}
+
 func (b generalBuilder) Build(ctx context.Context, pub string) error {
+	depsFiles := []string{
+		pub,
```

Comment:
I am not sure what `pub` is, could you please explain more about it?

Reply:
The pub key is filepath "/home/aseaday/.config/envd/id_rsa_envd.pub". I think It can be changed if users bootstrap again. So it need to be added into depsFiles.